modules/gapi/include/opencv2/gapi/rmat.hpp

```diff
@@ -88,7 +98,7 @@ class RMat
         // Implementation is responsible for setting the appropriate callback to
         // the view when accessed for writing, to ensure that the data from the view
         // is transferred to the device when the view is destroyed
-        virtual View access(Access) const = 0;
```

Comment:
Actually this needs to be non-const if we have an adapter with some host memory inside still we'll need to write it if accessed for write. My test adapters work only because of `cv::Mat::copyTo` declared as `const _OutputArray&(const _OutputArray&)`. If we move them to host own::Mat inside, this change is required